src/settings.rs

```diff
@@ -33,57 +28,36 @@ impl Default for Settings {
 impl Settings {
     pub const NONE_PROFILE_ID: &'static str = "none";
 
-    /// Opens a `FileChooserDialog` to select a folder and updates
+    /// Opens a `FileDialog` to select a folder and updates
     /// the settings with the selected folder.
     pub fn select_saving_location(&self, transient_for: Option<&impl IsA<gtk::Window>>) {
-        let chooser = gtk::FileChooserDialog::builder()
+        let dialog = gtk::FileDialog::builder()
             .modal(true)
-            .action(gtk::FileChooserAction::SelectFolder)
             .title(&gettext("Select Recordings Folder"))
+            .accept_label(&gettext("_Select"))
+            .initial_folder(&gio::File::for_path(self.saving_location()))
             .build();
-        chooser.set_transient_for(transient_for);
-
-        if let Err(err) =
-            chooser.set_current_folder(Some(&gio::File::for_path(self.saving_location())))
-        {
-            tracing::warn!("Failed to set current folder: {:?}", err);
-        }
-
-        chooser.add_button(&gettext("_Cancel"), gtk::ResponseType::Cancel);
-        chooser.add_button(&gettext("_Select"), gtk::ResponseType::Accept);
-        chooser.set_default_response(gtk::ResponseType::Accept);
-
-        chooser.present();
 
+        let parent = transient_for.unwrap().upcast_ref::<gtk::Window>();
         let inner = &self.0;
-        chooser.connect_response(clone!(@weak inner => move |chooser, response| {
-            if response != gtk::ResponseType::Accept {
-                chooser.close();
-                return;
-            }
-
-            let Some(directory) = chooser.file().and_then(|file| file.path()) else {
-                present_message(
-                    &gettext("No folder selected"),
-                    &gettext("Please choose a folder and try again."),
-                    Some(chooser),
-                );
-                return;
-            };
-
-            if !is_accessible(&directory) {
-                present_message(
-                    // Translators: {} will be replaced with a path to the folder.
-                    &gettext!("Cannot access “{}”", directory.display()),
-                    &gettext("Please choose an accessible location and try again."),
-                    Some(chooser),
-                );
-                return;
-            }
-
-            inner.set("saving-location", &directory).unwrap();
-            chooser.close();
-        }));
+        dialog.select_folder(
+            transient_for,
+            gio::Cancellable::NONE,
+            clone!(@weak inner, @weak parent => move |response| {
+                match response.ok().and_then(|file| file.path()) {
+                    Some(directory) => {
+                        inner.set("saving-location", &directory).unwrap();
+                    }
+                    None => {
```

Comment:
I tested it, but it seems to also show the `No folder selected` error dialog even when cancelled. 

I think we can do the following:
```rust
let transient_for = transient_for.as_ref();
let inner = &self.0;
dialog.select_folder(
    Some(transient_for),
    gio::Cancellable::NONE,
    clone!(@weak inner, @weak transient_for => move |response| {
        match response {
            Ok(folder) => {
                inner.set("saving-location", folder.path().unwrap()).unwrap();
            }
            Err(err) => {
                if err.matches(gtk::DialogError::Dismissed) || err.matches(gtk::DialogError::Cancelled) {
                    return;
                }

                present_message(
                    &gettext("Failed to select folder"),
                    &err.to_string(),
                    Some(&transient_for),
                );
            }
        }
    }),
);
```